proxy/sources/rawkuma.py

```diff
@@ -0,0 +1,197 @@
+from ..source.helpers import api_cache, get_wrapper
+
+class Rawkuma(ProxySource):
+    def get_reader_prefix(self):
+        return "rawkuma"
+
+    def shortcut_instantiator(self):
+        def handler(request, raw_url):
+            slug_name = self.get_slug_name(self.normalize_slug(raw_url))
+            canonical_chapter = self.parse_chapter(raw_url).replace(".", "-")
+            if canonical_chapter:
+                return redirect(
+                    f"reader-{self.get_reader_prefix()}-chapter-page",
+                    slug_name,
+                    canonical_chapter,
+                    "1",
+                )
+            else:
+                return redirect(
+                    f"reader-{self.get_reader_prefix()}-series-page",
+                    slug_name,
+                )
+
+        return [ re_path(r"^rk/(?P<raw_url>[\w\d\/:.-]+)", handler) ]
+
+    @staticmethod
+    def normalize_slug(denormal):
+        normal = re.split(r'-chapter-\d*', denormal.strip('/'))[0]
+        normal = 'https://' + re.sub(r'https?:\/\/', '', normal)
+        return normal
+
+    @staticmethod
+    def get_slug_name(url):
+        return url.strip("/").split("/")[-1]
+
+    @staticmethod
+    def get_series_url(meta_id):
+        return "https://rawkuma.com/manga/" + meta_id
+
+    @staticmethod
+    def parse_chapter(raw_url):
+        search = re.search(r'chapter-([\d-]+)$', raw_url.strip("/"))
+        if not search: return ""
+        return search.group(1).replace("-", ".")
+
+    def rk_scrape_common(self, meta_id):
+        series_url = self.get_series_url(meta_id)
+        resp = get_wrapper(series_url)
+        if resp.status_code == 200:
+            data = resp.text
+            soup = BeautifulSoup(data, "html.parser")
+            try:
+                title = soup.title.text.split('– Rawkuma')[0].strip()
+            except AttributeError:
+                return None
+            try:
+                author = "None"
+                artist = ""
+                for element in soup.select('div.infox > div > div.fmed > b'):
+                    if element.text == 'Author':
+                        author = element.next_sibling.next_sibling.text.strip()  # Need two next_siblings because one gives the '\n' between elements
+                    elif element.text == 'Artist':
+                        artist = element.next_sibling.next_sibling.text.strip()
+
+                if not artist: artist = author
+            except AttributeError:
+                author = "None"
+            try:
+                description = ""
+                paragraphs = soup.select_one('div.infox > div > div[itemprop="description"]').descendants
+                for element in paragraphs:
+                    if type(element) == NavigableString and element.parent.name != 'a':
+                        description += str(element)
+            except AttributeError:
+                description = "No description."
```

Comment:
nit: you can probably simplify these statements by declaring the defaults up-front rather than setting them in the rescue, then having these handlers try to parse more specific metadata from bs4. Would make this a bit easier to follow as well.

Reply:
How would this work without removing the try-except blocks?

Reply:
You can have both, the try-except block will catch the `AttributeError` and not set whatever value is being assigned if it fails. eg:

```python
title = "Default title"
description = "Default description"

try:
    title = "Set title"
except AttributeError:
    pass

try:
    def foo():
        raise AttributeError("foo")
    description = foo()
except AttributeError:
    pass

# Expected result:
# Set title Default description
print(title, description)
```

Alternatively, if you wanted to be creative you can have a method that takes a lambda or function and a default param so you can do something like:

```python
def try_parse(lambda, default):
    try:
        return lambda()
    except:
        return default

title = try_parse(lambda: soup.title.text.split('– Rawkuma')[0].strip(), "Default title")
description = try_parse(..., "Default description")
# etc
```

This is a weakly held opinion though, so feel free to ignore.